kpi-tracker/src/main.rs

```diff
@@ -85,32 +105,51 @@ struct ContractInstanceDetails {
     block_hash: BlockHash,
 }
 
+/// Represents a relation between an account and a transaction
+#[derive(Debug, Hash, PartialEq, PartialOrd, Ord, Eq)]
+struct TransactionAccountRelation(CanonicalAccountAddress, TransactionHash);
+
+/// Represents a relation between a contract and a transaction
+#[derive(Debug, Hash, PartialEq, PartialOrd, Ord, Eq)]
+struct TransactionContractRelation(ContractAddress, TransactionHash);
```

Comment:
I'm not sure if this is going to change when the database comes in, but it is almost always better to use named fields in structs. tuples are only convenient when writing them :)